core/src/main/scala/kafka/server/BrokerRaftEventManager.scala

```diff
@@ -0,0 +1,165 @@
+package kafka.server
+
+import java.util.concurrent.{LinkedBlockingQueue, TimeUnit}
+
+import kafka.coordinator.group.GroupCoordinator
+import kafka.metrics.KafkaMetricsGroup
+import kafka.server.QuotaFactory.QuotaManagers
+import kafka.utils.ShutdownableThread
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.internals.FatalExitError
+import org.apache.kafka.common.message.UpdateMetadataRequestData
+import org.apache.kafka.common.network.ListenerName
+import org.apache.kafka.common.protocol.{ApiKeys, ApiMessage}
+import org.apache.kafka.common.requests.UpdateMetadataRequest
+import org.apache.kafka.common.utils.Time
+import org.apache.kafka.raft.RaftMessage
+
+object BrokerRaftEventManager {
+  val BrokerRaftEventThreadName = "broker-raft-event-thread"
+  val EventQueueTimeMetricName = "EventQueueTimeMs"
+  val EventQueueSizeMetricName = "EventQueueSize"
+}
+
+trait RaftEventProcessor {
```

Comment:
nit: you might consider replacing "raft" in these names with "metadata." I think the fact that we use raft is more of an implementation detail. More important is what the events represent.